src/charm.py

```diff
@@ -119,6 +153,10 @@ def _on_collect_unit_status(self, event: CollectStatusEvent):
             logger.info("Waiting for statefulset to be patched")
             return
         self.unit.set_workload_version(self._get_workload_version())
+        if not self._kubernetes_multus.is_ready():
```

Comment:
Probably not critical, but how about making sure that Multus is ready before checking for the N2 relation?